x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/JobManager.java

```diff
@@ -466,26 +478,40 @@ private static void appendCommaSeparatedSet(Set<String> items, StringBuilder sb)
         sb.append("]");
     }
 
-    public void updateProcessOnCalendarChanged(List<String> calendarJobIds) {
+    public void updateProcessOnCalendarChanged(List<String> calendarJobIds, ActionListener<Boolean> updateListener) {
```

Comment:
I think this was already expected to work in an async fashion.  It was just queueing the update but the callers are not really expecting the update to complete before they respond. Unless I'm missing something, we probably can remove the listener here and simple have an internal one for doing the callback after we expand the job ids.